src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2832,9 +2832,9 @@ ErrorIfContainsUnsupportedSubquery(MultiNode *logicalPlanNode,
 	/* currently in the planner we only allow one subquery in from-clause*/
 	Assert(list_length(subqueryMultiTableList) == 1);
 
-	restrictionEquivalenceForPartitionKeys =
-		RestrictionEquivalenceForPartitionKeys(plannerRestrictionContext);
-	if (!restrictionEquivalenceForPartitionKeys)
+	safeToPushdownSubquery =
+		SafeToPushDownSubquery(plannerRestrictionContext);
+	if (!safeToPushdownSubquery)
```

Comment:
I don't like this generic error message. We should differentiate unions and joins, at least,